src/main/java/com/vaadin/lazyloadwrapper/widgetset/client/ui/LazyLoadWrapperConnector.java

```diff
@@ -140,6 +148,11 @@ private void updateToThisLLW() {
             if (getState().clientSideIsVisible && getChildren().size() > 0
                     && getState().mode != MODE_LAZY_LOAD_DRAW) {
                 ComponentConnector childConnector = getChildComponents().get(0);
+				
+                final String caption = childConnector.getState().caption;
+                if (caption != null && !"".equals(caption)) {
```

Comment:
Is this code executed only once? What if caption actually changes from non-empty to empty? Currently, caption would be still visible.